USFMToolsSharp/HtmlRenderer.cs

```diff
@@ -2,38 +2,90 @@
 using System.Collections.Generic;
 using System.Text;
 using USFMToolsSharp.Models.Markers;
+using USFMToolsSharp.Models;
 
 namespace USFMToolsSharp
 {
     public class HtmlRenderer
     {
         public List<string> UnrenderableTags;
+        public List<string> FootnoteTextTags;
+        public HTMLConfig ConfigurationHTML;
+        
+
+        public string FrontMatterHTML { get; set; }
+        public string InsertedFooter { get; set;}
+        public string InsertedHead { get; set; }
 
         public HtmlRenderer()
         {
             UnrenderableTags = new List<string>();
+            FootnoteTextTags = new List<string>();
+
+            ConfigurationHTML = new HTMLConfig();
+        }
+        public HtmlRenderer(HTMLConfig config)
+        {
+
+            ConfigurationHTML = config;
+
+            UnrenderableTags = new List<string>();
+            FootnoteTextTags = new List<string>();
+
         }
 
         public string Render(USFMDocument input)
         {
             UnrenderableTags = new List<string>();
-            string encoding = GetEncoding(input);
+            var encoding = GetEncoding(input);
             StringBuilder output = new StringBuilder();
             output.AppendLine("<html>");
-            output.AppendLine("<head>");
-            if (!string.IsNullOrEmpty(encoding))
+
+            if (InsertedHead == null)
+            {
+                output.AppendLine("<head>");
+                if (!string.IsNullOrEmpty(encoding))
+                {
+                    output.Append($"<meta charset=\"{encoding}\">");
+                }
+                output.AppendLine("<link rel=\"stylesheet\" href=\"style.css\">");
+
+                output.AppendLine("</head>");
+            }
+            else
             {
-                output.Append($"<meta charset=\"{encoding}\">");
+                output.AppendLine(InsertedHead);
             }
-            output.AppendLine("<link rel=\"stylesheet\" href=\"style.css\">");
-            output.AppendLine("</head>");
+
             output.AppendLine("<body>");
 
-            foreach(Marker marker in input.Contents)
+            output.AppendLine(FrontMatterHTML);
+
+            // HTML tags can only have one class, when render to docx
+
+            foreach(string class_names in ConfigurationHTML.divClasses)
+            {
+                if(class_names.Length>0)

```

Comment:
And one more thing you don't need this if statement here.

Reply:
Agreed.  We know class_names has a nonzero length because we're inside the `for` loop.